src/fidesops/api/v1/endpoints/dataset_endpoints.py

```diff
@@ -161,6 +176,26 @@ def patch_datasets(
     )
 
 
+def __validate_saas_dataset(
+    connection_config: ConnectionConfig, dataset: FidesopsDataset
+) -> None:
+    if connection_config.saas_config is None:
+        raise SaaSConfigNotFoundException(
+            f"Connection config '{connection_config.key}' must have a "
+            "SaaS config before adding a dataset"
+        )
+    else:
+        for collection in dataset.collections:
+            for field in collection.fields:
+                graph_field = to_graph_field(field)
+                if graph_field.references or graph_field.identity:
+                    raise ValidationError(
+                        "A dataset for a ConnectionConfig type of saas is not "
+                        "allowed to have references or identities. Please add "
+                        "them to the SaaS config."
+                    )
```

Comment:
Nice validation here - related, a leftover comment from your other PR - `SaaSConfig.get_graph` separately checks if there are references defined and if there is an identity defined, and if either case happens, a separate ScalarField is added.  https://github.com/ethyca/fidesops/pull/179#discussion_r807271127

1) I was wondering if someone could define a single field as both an identity, and a referenced field. If they made that mistake on the saas config, they would have two ScalarFields created.

2) Are we handling object and array fields with SaasConfigs yet? It seems like SaasConfig.get_graph can only create ScalarFields.

Reply:
Added a validation to ConnectorParam which throws a `ValueError` if both `references` and `identities` are defined for the same param.

Reply:
As for the object and array fields, we can keep using scalar fields for the SaaS config since all we need is a place to hold references and identities to be able to add them to the dataset.